### PR TITLE
Add doc to Regex::Options

### PR DESCRIPTION
**Pull request by jan-zajic**
So at least Crystal lang contributors does not have to search in PCRE docs.

### CODE REVIEW

File: src/regex.cr [hunk 1]
```diff
@@ -212,14 +213,17 @@ class Regex
...
     # mapping `MULTILINE` to `PCRE_MULTILINE | PCRE_DOTALL`.
     MULTILINE = 6
-    EXTENDED  = 8
-    # :nodoc:
+    # Ignore white space and # comments.
+    EXTENDED = 8
+    # Force pattern anchoring.
     ANCHORED = 16
-    # :nodoc:
+    # Run in UTF-8 mode
```
Comment (asterite) [inline src/regex.cr]: This isn't public and it's just used internally

File: src/regex.cr [hunk 2]
```diff
@@ -212,14 +213,17 @@ class Regex
...
     UTF_8 = 0x00000800
-    # :nodoc:
+    # Do not check the pattern for UTF-8
+    # validity (only relevant if
+    # PCRE_UTF8 is set)
     NO_UTF8_CHECK = 0x00002000
```
Comment (asterite) [inline src/regex.cr]: No need to document thks

File: src/regex.cr [hunk 3]
```diff
@@ -212,14 +213,17 @@ class Regex
...
-    # :nodoc:
+    # Allow duplicate names for subpatterns.
     DUPNAMES = 0x00080000
```
Comment (asterite) [inline src/regex.cr]: It was nodoc for a reason

**Comment by jan-zajic** [comment]
@asterite fixed

File: src/regex.cr [hunk 4]
```diff
@@ -203,6 +203,7 @@ class Regex
 
   @[Flags]
   enum Options
+    # Case insensitive match
```
Comment (straight-shoota) [inline src/regex.cr]:
```suggestion
    # Case insensitive match.
```

File: src/regex.cr [hunk 5]
```diff
@@ -212,8 +213,9 @@ class Regex
     # flag that activates both behaviours, so here we do the same by
     # mapping `MULTILINE` to `PCRE_MULTILINE | PCRE_DOTALL`.
     MULTILINE = 6
-    EXTENDED  = 8
-    # :nodoc:
+    # Ignore white space and # comments.
```
Comment (straight-shoota) [inline src/regex.cr]:
```suggestion
    # Ignore white space and `#` comments.
```

**Pull request merged**